fvm/Equation/Operator.cpp

```diff
@@ -14,7 +14,9 @@ using namespace std;
 /**
  * Constructor.
  */
-Operator::Operator(Grid *grid) : grid(grid) {}
+Operator::Operator(Grid *grid) : grid(grid) {
+vectorElementsSingleTerm=new real_t(this->grid->GetNCells());
```

Comment:
I have also thought about point 3, but since the macros in OtherQuantityHandle.cpp want functions that return the data to be stored instead of void-functions with an output array as an input parameter, I implemented it this way to be compatible with that (although that does, of course, not mean that it is the best way to do it).